src/test_logging.c

```diff
@@ -0,0 +1,301 @@
+/*
+  This file is part of Restraint.
+
+  Restraint is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Restraint is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Restraint.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <glib.h>
+
+#define LOG_MANAGER_DIR "./test_logging_logs"
+
+#include "logging.c"
+
+static int message_callback_calls = 0;
+
+static void
+check_log_file_contents (const RstrntTask *task,
+                         const char       *child,
+                         const char       *contents)
+{
+    g_autofree char *path = NULL;
+    g_autoptr (GFile) log_directory_file = NULL;
+    g_autoptr (GFile) child_file = NULL;
+    g_autofree char *child_path = NULL;
+    g_autoptr (GMappedFile) mapped_file = NULL;
+    size_t file_length;
+    char *file_contents;
+    size_t contents_length;
+
+    path = g_build_path ("/", LOG_MANAGER_DIR, task->task_id, NULL);
+    log_directory_file = g_file_new_for_path (path);
+    child_file = g_file_get_child (log_directory_file, child);
+    child_path = g_file_get_path (child_file);
+    mapped_file = g_mapped_file_new (child_path, FALSE, NULL);
+    file_length = g_mapped_file_get_length (mapped_file);
+    file_contents = g_mapped_file_get_contents (mapped_file);
+    contents_length = strlen (contents);
+
+    g_assert_cmpuint (contents_length, ==, file_length);
+    g_assert_cmpstr (contents, ==, file_contents);
+}
+
+static void
+test_rstrnt_log_write (gconstpointer user_data)
+{
+    const RstrntTask *task;
+    g_autofree char *task_contents = NULL;
+    g_autofree char *harness_contents = NULL;
+
+    task = user_data;
+    task_contents = g_strdup_printf ("el task");
+    harness_contents = g_strdup_printf ("el harness");
+
+    rstrnt_log (task, RSTRNT_LOG_TYPE_TASK, "%s", task_contents);
```

Comment:
Your test code is the only place that uses rstrnt_log.  Might it make more sense to test rstrnt_log_bytes() which is the function used by restraint?

Reply:
Both are wrapping `rstrnt_log_manager_append_to_log`, and you can see that `rstrnt_log` is taking care of the boilerplate needed to test `rstrnt_log_bytes`. At this point testing `rstrnt_log_manager_append_to_log` should be enough.